install.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+set -e
+
+REPO="GoogleCloudPlatform/kubectl-ai"
+BINARY="kubectl-ai"
+
+# Detect OS
+OS="$(uname | tr '[:upper:]' '[:lower:]')"
+case "$OS" in
+  linux)   OS="linux" ;;
+  darwin)  OS="darwin" ;;
+  *)
+    echo "If you are on Windows or another unsupported OS, please follow the manual installation instructions at:"
+    echo "https://github.com/GoogleCloudPlatform/kubectl-ai#manual-installation"
+    exit 1
+    ;;
+esac
+
+# Detect ARCH
+ARCH="$(uname -m)"
+case "$ARCH" in
+  x86_64|amd64) ARCH="amd64" ;;
+  arm64|aarch64) ARCH="arm64" ;;
+  *)
+    echo "If you are on an unsupported architecture, please follow the manual installation instructions at:"
+    echo "https://github.com/GoogleCloudPlatform/kubectl-ai#manual-installation"
+    exit 1
+    ;;
+esac
+
+# Get latest version tag from GitHub API (portable, no grep -P)
+LATEST_TAG=$(curl -s "https://api.github.com/repos/$REPO/releases/latest" | sed -n 's/.*"tag_name": *"\([^"]*\)".*/\1/p')
+if [ -z "$LATEST_TAG" ]; then
+  echo "Failed to fetch latest release tag."
+  exit 1
+fi
+
+# Compose download URL
+TARBALL="kubectl-ai_${OS}_${ARCH}.tar.gz"
```

Comment:
One thing I noticed that the OS name in the tar file is capitalized `Darwin`, `Linux`, `Windows`, but here we will be using OS name all lower-cased. I wonder if it will work with curl on all OSes. Lets try to match them exact to avoid any edge cases.

I think this script is super important because failures here will frustrate the users and some of them may never bother to install and just abort it. So better to be safe than sorry.

Reply:
I agree. While URLs are case-insensitive, it would be a good idea to make them 100% matched just in case.